### PR TITLE
Remove unused gutenboarding script

### PR DESCRIPTION
**Pull request by noahtallen**
## Proposed Changes
I think this script has been unused for a while. It was added for some aspects of gutenboarding, which is no longer used (#74475). This lets us remove a dependency on `sharp`, which has been problematic in the past (#55292). And I noticed it causing issues with the latest typescript update, so it'd be nice to remove it.

## Testing Instructions
None; CI should pass. Also verify `sharp` isn't imported elsewhere.

### CODE REVIEW

File: bin/install-gutenboarding-design-thumbnails-dependencies.js [hunk 1]
```diff
@@ -1,48 +0,0 @@
...
-	process.exit( installResult.status );
-}
-
-/*
- * Add screenshot dependencies
- * I didn't want to add these to dependencies just for this script since these are kinda heavy ones)
- * https://www.npmjs.com/package/capture-website
- * https://www.npmjs.com/package/sharp
- */
-const installDependencies = spawnSync( 'yarn', [ 'add', '-W', 'capture-website', 'sharp' ], {
```
Comment (tyxla) [inline bin/install-gutenboarding-design-thumbnails-dependencies.js:38]: While we're here, I think we can remove the `capture-website` dependency, too. This seems to be its sole usage.

**Review by tyxla — APPROVED** [review]
Looking great! 🚀 

I'd only suggest removing another unused package before shipping.

File: test/e2e/docs/troubleshooting.md [hunk 2]
```diff
@@ -40,24 +40,3 @@ yarn install
 ```
 
 Once complete, running `git commit` should no longer trigger the git pre-commit hook error.
-
-## Package 'lcms2', required by 'vips', not found
```
Comment (noahtallen) [inline test/e2e/docs/troubleshooting.md:44]: This won't be an issue any more :D

File: bin/install-gutenboarding-design-thumbnails-dependencies.js (hunk 1 above)
Comment (noahtallen) [reply bin/install-gutenboarding-design-thumbnails-dependencies.js:38]: done!

**Pull request merged**